mcs/class/corlib/System/TimeZoneInfo.Unity.cs

```diff
@@ -0,0 +1,136 @@
+//
+// System.TimeZoneInfo helper for MonoTouch
+// 	because the devices cannot access the file system to read the data
+//
+// Authors:
+//	Sebastien Pouliot  <sebastien@xamarin.com>
+//
+// Copyright 2011-2013 Xamarin Inc.
+//
+// The class can be either constructed from a string (from user code)
+// or from a handle (from iphone-sharp.dll internal calls).  This
+// delays the creation of the actual managed string until actually
+// required
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+#if UNITY
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using System.IO;
+
+namespace System {
+
+	public partial class TimeZoneInfo {
+		enum TimeZoneData
+		{
+		    DaylightSavingStartIdx,
+		    DaylightSavingEndIdx,
+		    UtcOffsetIdx,
+		    AdditionalDaylightOffsetIdx
+		};
+		
+		enum TimeZoneNames
+		{
+		    StandardNameIdx,
+		    DaylightNameIdx
+		};
+		static TimeZoneInfo CreateLocalUnity ()
+		{
+			Int64[] data;
+		    string[] names;
+		    if (!System.CurrentSystemTimeZone.GetTimeZoneData (1970, out data, out names))
+				throw new NotSupportedException ("Can't get timezone name.");
+
+			TimeSpan utcOffsetTS = TimeSpan.FromTicks(data[(int)TimeZoneData.UtcOffsetIdx]);
+		    char utcOffsetSign = (utcOffsetTS >= TimeSpan.Zero) ? '+' : '-';
+		    string displayName = "(GMT" + utcOffsetSign + utcOffsetTS.ToString(@"hh\:mm") + ") Local Time";
+		    string standardDisplayName = names[(int)TimeZoneNames.StandardNameIdx];
+		    string daylightDisplayName = names[(int)TimeZoneNames.DaylightNameIdx];
+		    
+		    //Create The Adjustment Rules For This TimeZoneInfo.
+		    var adjustmentList = new List<TimeZoneInfo.AdjustmentRule>();
+		    for(int year = 1970; year <= 2037; year++)
```

Comment:
You should change this to 1973. the android mktime barfs before then

Reply:
Also we need to make sure that IOS doesn't have some similar weird issue. 1970 forward worked fine on desktop...

Reply:
We should probably add a basic runtime test that verifys that utc != localtime on all our platforms